src/components/input/input.tsx

```diff
@@ -32,6 +32,7 @@ import { createObserver } from "../../utils/observers";
 import { InteractiveComponent, updateHostInteraction } from "../../utils/interactive";
 
 type NumberNudgeDirection = "up" | "down";
+type SetValueOrigin = "initial" | "loading" | "user" | "direct";
```

Comment:
Yes, exactly!  This was my initial stab at naming these origins.  I like `direct` and that is echoed in places like MDN for direct value setting.  Either `user` or `user-interaction` works, but I could even see going so far as making user interaction sources more specific to use words like `keyboard` or `mouse` even, depending on the component.  

Normal typing would be handled by the `keyboard` origin, and I can imagine a use case like the number input where a user clicks the up/down arrows with the mouse and differentiating between `mouse` interaction that way.

Reply:
Yeah it would be great if these could be cleaned up into one private property and standardize the language.

It seems like these could be grouped..

```ts
  private previousEmittedValue: string;

  private previousValue: string;

  private previousValueOrigin: SetValueOrigin = "initial";

  private userChangedValue = false;
```

Reply:
Perfect, yeah, this is just the kind of feedback I'm hoping for.  I'll refer back to this example in next week's meeting on the topic.

Reply:
Another thing would be if the arguments on setValue were more descriptive of what it is going to do. Like if it should emit an event, commit the changes, format the value, etc. IMO the origin logic is kind of confusing.

I'm not sure why we are emitting the native event inside of the custom event either. Is that a pattern/practice we want to do?